File: 0565_array_nesting/array_ nesting.java
class Solution {
    // Time: O(n)
    // Space: O(1)
    public int arrayNesting(int[] nums) {
        int res = 0;
        for (int i = 0; i < nums.length; i++) {
            int count = 0;
            while (nums[i] != -1) {
                int tmp = nums[i];
                nums[i] = -1;
                i = tmp;
                count++;
            }
            res = Math.max(res, count);
        }
        return res;
    }
}

File: 0565_array_nesting/array_nesting.py
from typing import List


def arrayNesting(nums: List[int]) -> int:
    """
    Time: O(n)
    Space: O(1)
    """
    res = 0
    for i in range(len(nums)):
        cnt = 0
        while nums[i] != -1:
            tmp = nums[i]
            nums[i] = -1
            cnt += 1
            i = tmp
        res = max(res, cnt)
    return res